src/tabs/newTab.tsx

```diff
@@ -1,16 +1,67 @@
+import { useEffect, useState } from "react"
+
+import "./NewTab.css"
+
+import { BlocksRenderer } from "@strapi/blocks-react-renderer"
+import Browser from "webextension-polyfill"
+
+import { SlimPopup } from "~types/eventTypes"
+
 const NewTab = () => {
   const params = new URLSearchParams(window.location.search)
-  const width = params.get("width")
-  const height = params.get("height")
-  const url = "http://localhost:1337" + params.get("url")
-
+  const index = params.get("index")
+  const [popup, setPopup] = useState<SlimPopup>()
+  useEffect(() => {
+    const getFromStorage = async () => {
+      const { popups } = await Browser.storage.session.get(["popups"])
```

Comment:
this could not get noticed for a long time but just as a general best practice we should name keys going into storage in a more 'unique' way. "popup" is fairly generic and we run the risk of our storage being overridden.

Again, it's a fairly reduced risk because of how we're using this in a single session and all that, but just one of those that could lead to a super edge issue that would take for ever to uncover if it happened